internal/admission/validator.go

```diff
@@ -630,3 +632,55 @@ func (m *managerClientConsumerGetter) ListAllConsumers(ctx context.Context) ([]k
 	}
 	return consumers.Items, nil
 }
+
+func (validator KongHTTPValidator) ValidateCustomEntity(ctx context.Context, entity kongv1alpha1.KongCustomEntity) (bool, string, error) {
+	// If the spec.contollerName does not match the ingress class name,
+	// ignore it as it is not controlled by the controller.
+	// REVIEW: should we also match the "ingress class" annotation here?
+	if !validator.ingressClassMatcher(&metav1.ObjectMeta{
+		Annotations: map[string]string{
+			annotations.IngressClassKey: entity.Spec.ControllerName,
+		},
+	}, annotations.IngressClassKey, annotations.ExactClassMatch) {
+		validator.Logger.V(util.DebugLevel).Info("Skipped validation because the controller name does not match", "controller_name", entity.Spec.ControllerName)
+		return true, "", nil
+	}
+
+	// REVIEW: should we validate group/kind of `spec.parentRef` here because we only want it to be attached to plugins?
+
+	fields, err := kongstate.RawConfigToConfiguration(entity.Spec.Fields.Raw)
+	if err != nil {
+		return false, fmt.Sprintf(ErrTextCustomEntityFieldsUnmarshalFailed, err), nil
+	}
+
+	schemaService, hasClient := validator.AdminAPIServicesProvider.GetSchemasService()
+	// Skip validation on Kong gateway if we do not have available client.
+	if !hasClient {
+		validator.Logger.V(util.DebugLevel).Info("Skipped because no schema service available")
```

Comment:
This and other logger calls should mention the entity namespace/name they're processing. Absent that there's no indication of whatever was being processed when the event occurred.

Probably just make a `validator.Logger.WithValues(...)` child logger at the start of `ValidateCustomEntity()`. I don't think there's any log where we'd want to omit that info.

Reply:
Added a logger.